src/utils/validationFunctions.js

```diff
@@ -18,3 +18,16 @@ export function register(email, password) {
     throw Error(ERR.PASSWORD_REQUIREMENTS)
   }
 }
+
+export function validateCanPatch(req) {
+  const { role } = req.user
+  const { id } = req.body
+  const targetUserId = Number(req.params.id)
+  const isUser = id === targetUserId
+  const isTeacher = role === 'TEACHER'
+
+  if (!isTeacher && !isUser) {
+    return false
+  }
+  return true
+}
```

Comment:
Alright, I understand the teacher requirement now. However, I'm not sure this is a secure way to handle the validation here. All this is doing is comparing the `id` in the body with the `id` in the parameters to check that the user is updating their own data, this isn't accurate. I could send a request to update your details if I provide your user ID in both instances.

I think you want to get the `id` from the user attached to the request, e.g. `isUser = id === req.user.id`

Reply:
That was an oversight on my part - I have updated the code so that the id is destructured from req.user